From c7f3e2a8fe530beec6103cb9071ccc41458879aa Mon Sep 17 00:00:00 2001
From: Tony Tascioglu <tony.tascioglu@windriver.com>
Date: Mon, 21 Jun 2021 12:34:22 -0400
Subject: [PATCH] libyang: fix initial build errors

This patch addresses build errors seen when integrating libyang as a
recipe.

The call to pcre2-config has been commented out. *-config calls are
being replaced by pkgconfig, and we want to avoid the binary config
calls. This call can safely be discarded since it is used to check
the version of libpcre2 on the system, and all recent YP builds meet
the minimum required version (v10.21).

There is a missing include statement for stdint.h in utests.h which
causes build problems. stdint.h is required to be imported by cmocka.h
prior to the cmocka file. Adding the stdint allows it to build the
tests correctly

Upstream-Status: Pending

Signed-off-by: Tony Tascioglu <tony.tascioglu@windriver.com>
---
 CMakeModules/FindPCRE2.cmake | 2 +-
 tests/utests/utests.h        | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/CMakeModules/FindPCRE2.cmake b/CMakeModules/FindPCRE2.cmake
index 8f60daa2..ad014f14 100644
--- a/CMakeModules/FindPCRE2.cmake
+++ b/CMakeModules/FindPCRE2.cmake
@@ -12,7 +12,7 @@ FIND_PATH(PCRE2_INCLUDE_DIR pcre2.h)
 FIND_LIBRARY(PCRE2_LIBRARY NAMES libpcre2.a pcre2-8)
 
 # Check required version
-execute_process(COMMAND pcre2-config --version OUTPUT_VARIABLE PCRE2_VERSION OUTPUT_STRIP_TRAILING_WHITESPACE)
+# execute_process(COMMAND pcre2-config --version OUTPUT_VARIABLE PCRE2_VERSION OUTPUT_STRIP_TRAILING_WHITESPACE)
 
 # Handle the QUIETLY and REQUIRED arguments and set PCRE_FOUND to TRUE if all listed variables are TRUE.
 INCLUDE(FindPackageHandleStandardArgs)
diff --git a/tests/utests/utests.h b/tests/utests/utests.h
index 5bea0459..a256120c 100644
--- a/tests/utests/utests.h
+++ b/tests/utests/utests.h
@@ -23,6 +23,7 @@
 #include <stddef.h>
 #include <stdlib.h>
 
+#include <stdint.h>
 #include <cmocka.h>
 
 #include <string.h>
-- 
2.32.0

